lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -2133,14 +2133,21 @@ namespace {
         for (auto arg : *call->getArgs()) {
           if (auto declRef = dyn_cast<DeclRefExpr>(arg.getExpr())) {
             if (auto var = dyn_cast<VarDecl>(declRef->getDecl())) {
-              if (var->isSelfParameter()) {
-                return var;
-              }
+              return var;
```

Comment:
This is somewhat of a surprise. Do we not need to check for `self` anywhere?

Reply:
Explained below, it is subsumed by checking "is known local?", which `self` just happens to be 👍